ts/session/objects/ClosedGroup.ts

```diff
@@ -0,0 +1,431 @@
+// This is the (Closed | Medium) Group equivalent to the SessionGroup type.
+
+import { ClosedGroupType, PubKey } from '../types';
+import { UserUtil } from '../../util';
+import { MultiDeviceProtocol } from '../protocols';
+import { onGroupReceived } from '../../receiver/receiver';
+import * as Data from '../../../js/modules/data';
+
+import { createSenderKeysForMembers } from '../medium_group';
+import { GroupUtils, StringUtils } from '../utils';
+import { Constants, Utils } from '..';
+import { ConversationModel } from '../../../js/models/conversations';
+import * as _ from 'lodash';
+
+interface ClosedGroupParams {
+  id: PubKey;
+  type: ClosedGroupType;
+  name: string;
+  admins: Array<PubKey>;
+  members: Array<PubKey>;
+  expireTimer: number;
+}
+
+interface ClosedGroupUpdateParams {
+  // Closed groups do not currently support avatars
+  name: string;
+  avatar?: string;
+  recipients: Array<PubKey>;
+  members: Array<PubKey>;
+  isMediumGroup: boolean;
+  options: any;
+}
+
+export class ClosedGroup {
+  public readonly id: PubKey;
+  public readonly type: ClosedGroupType;
+  private _name: string;
```

Comment:
Why the "_name" notation? I know this is sometimes done in other languages like C++, but the main reason is that `this.` part is implicit there, so `_` can at least indicate that we work with a member. I'm not sure we need this in JS.